kotlin/android/app/src/main/java/dev/firezone/android/features/session/ui/SessionActivity.kt

```diff
@@ -72,9 +74,15 @@ class SessionActivity : AppCompatActivity() {
         super.onDestroy()
     }
 
-    fun onViewResourceToggled(resourceToggled: ViewResource) {
-        Log.d(TAG, "Resource toggled $resourceToggled")
-        tunnelService?.resourceToggled(resourceToggled)
+    fun internetState(): ResourceState {
+        return tunnelService?.internetState() ?: ResourceState.UNSET
+    }
+
+    private fun onInternetResourceToggled(): ResourceState {
+        tunnelService?.internetResourceToggled(internetState().toggle())
+        refreshList()
+        Log.d(TAG, "Internet resource toggled ${internetState()}")
```

Comment:
Next two lines only make sense to execute if the tunnelService is valid, no?

```suggestion
        tunnelService?.let {
          it.internetResourceToggled(internetState().toggle())
          refreshList()
          Log.d(TAG, "Internet resource toggled ${internetState()}")
        }
```

Reply:
is there any point at which this can actually happen to be nil? Maybe I just use `!!`?